pkg/jobs/jobs_test.go

```diff
@@ -2794,7 +2781,7 @@ func TestStartableJobTxnRetry(t *testing.T) {
 		txn.KV().SetDebugName(txnName)
 		return jr.CreateStartableJobWithTxn(ctx, &sj, jobID, txn, rec)
 	}))
-	require.True(t, haveInjectedRetry)
+	verifyFunc()
```

Comment:
nit: maybe instead of `verifyFunc()`, let the caller decide what to do on failure, something like `require.True(t, haveInjectedRetry())`?